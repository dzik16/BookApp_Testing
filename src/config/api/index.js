export * from './action';
export * from './reducer';
export * from './store';
export * from './types';
