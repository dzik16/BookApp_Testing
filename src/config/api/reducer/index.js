export * from './userReducer';
export * from './bookReducer';
export * from './globalReducer';
