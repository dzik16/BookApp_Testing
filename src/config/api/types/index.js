// user
export const REGISTER = '@REGISTER';
export const REGISTER_SUCCESS = '@REGISTER_SUCCESS';
export const LOGIN = '@LOGIN';
export const LOGOUT = '@LOGOUT';

// book
export const BOOKS_RECOMMENDED = '@BOOKS_RECOMMENDED';
export const BOOKS_POPULAR = '@BOOKS_POPULAR';
export const BOOKS_ID = '@BOOKS_ID';

// global
export const LOADING = '@LOADING';
export const REFRESH = '@REFRESH';
export const INTERNET = '@INTERNET';
