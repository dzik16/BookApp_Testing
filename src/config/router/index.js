/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import Router from './Stack';
import { navigationRef } from './navigate';

const index = () => (
  <NavigationContainer ref={navigationRef}>
    <Router />
  </NavigationContainer>
);

export default index;
