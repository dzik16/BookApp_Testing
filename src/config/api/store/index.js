import AsyncStorage from '@react-native-async-storage/async-storage';
import { applyMiddleware, combineReducers, createStore } from 'redux';
import reduxLogger from 'redux-logger';
import { persistReducer, persistStore } from 'redux-persist';
import ReduxThunk from 'redux-thunk';
import { authReducers, dataReducers, globalReducers } from '../reducer';

const persistConfig = {
  key: 'root',
  blacklist: ['dataBooks'],
  storage: AsyncStorage,
};

const rootReducer = {
  dataBooks: dataReducers,
  Auth: authReducers,
  global: globalReducers,
};

const configPersist = persistReducer(
  persistConfig,
  combineReducers(rootReducer),
);

export const Store = createStore(
  configPersist,
  applyMiddleware(ReduxThunk, reduxLogger),
);

export const Persistore = persistStore(Store);
