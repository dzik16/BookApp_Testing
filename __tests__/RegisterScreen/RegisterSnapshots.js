import {
    shallow, configure,
} from 'enzyme';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { authReducers, dataReducers } from '../../src/config/api';
import Register from '../../src/screens/auth/RegisterScreen';

jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter');

jest.mock('react-native-flash-message', () => "FlashMessageManager");

jest.mock('@react-native-async-storage/async-storage', () => "AsyncStorage");

jest.mock('react-native-share', () => ({
}));

configure({ adapter: new Adapter(), disableLifecycleMethods: true });

describe('Register Screen test', () => {
    const initialState = {
        dataBooks: dataReducers,
        Auth: authReducers,
    };

    const mockStore = configureStore();
    const store = mockStore(initialState);
    const registerWrapper = shallow(
        <Provider store={store}>
            <Register />
        </Provider>,
    );

    jest.mock('@react-navigation/native', () => ({
        useNavigation: (component) => component,
    }));

    it('should renders `Register Screen` module correctly', () => {
        expect(registerWrapper).toMatchSnapshot();
    });
    describe('Check component', () => {
        it('should create find `Input`', () => {
            expect(registerWrapper.find('Input').exists());
        });

        it('should create `TouchableOpacity` component', () => {
            expect(registerWrapper.find('TouchableOpacity').exists());
        });
        it('should create `LinkComponent` component', () => {
            expect(registerWrapper.find('LinkComponent').exists());
        });
    });
});
