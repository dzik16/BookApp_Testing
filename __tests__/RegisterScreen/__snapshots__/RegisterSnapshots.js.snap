// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Screen test should renders \`Register Screen\` module correctly 1`] = `ShallowWrapper {}`;
